sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_delete_test_live.cpp

```diff
@@ -320,3 +319,84 @@ TEST_F(KeyVaultClientTest, GetDeletedKey)
         Azure::Security::KeyVault::Keys::KeyType::KeyTypeToString(deletedKey.Key.KeyType));
   }
 }
+
+TEST_F(KeyVaultClientTest, DeleteOperationResumeToken)
+{
+  Azure::Security::KeyVault::Keys::KeyClient keyClient(m_keyVaultUrl, m_credential);
+  auto keyName = GetUniqueName();
+
+  {
+    auto keyResponse
+        = keyClient.CreateKey(keyName, Azure::Security::KeyVault::Keys::JsonWebKeyType::Ec);
+    CheckValidResponse(keyResponse);
+    auto keyVaultKey = keyResponse.Value;
+    EXPECT_EQ(keyVaultKey.Name(), keyName);
+  }
+  std::string resumeToken;
+  {
+    auto keyResponseLRO = keyClient.StartDeleteKey(keyName);
+    resumeToken = keyResponseLRO.GetResumeToken();
+  }
+  // Resume operation from token
+  {
+    auto resumeOperation = keyClient.ResumeDeleteKey(resumeToken);
+    resumeOperation.PollUntilDone(std::chrono::milliseconds(500));
+  }
+  {
+    // recover
+    auto recoverOperation = keyClient.StartRecoverDeletedKey(keyName);
```

Comment:
Out of curiosity, what happens if I pass in the resume token from `StartDeleteKey` (i.e. `keyResponseLRO.GetResumeToken();`) into `ResumeRecoverDeletedKey`

Reply:
The operation would be constructed as Failed.

Reply:
Could we detect that and throw instead?

Does `StartDeleteKey` ever return an Operation in the failed state? If not, we way want to avoid that in the `Resume*` case.